integration/upload_test_util.go

```diff
@@ -0,0 +1,198 @@
+	return requestedFiles
+}
+
+// StartUploads starts the uploads for all given file upload requests
+func (suite *FileUploadSuite) StartUploads(TestUpload Upload, featureID string, requestedFiles map[string]string) {
+	pathLastUpload := util.GetFeaturePropertyPath(featureID, propertyLastUpload)
+	topicCreated := util.GetTwinEventTopic(suite.ThingCfg.DeviceID, protocol.ActionCreated)
+	topicModified := util.GetTwinEventTopic(suite.ThingCfg.DeviceID, protocol.ActionModified)
+
+	connEvents, err := util.NewDigitalTwinWSConnection(suite.Cfg)
+	require.NoError(suite.T(), err, msgFailedCreateWebsocketConnection)
+	defer connEvents.Close()
+
+	err = util.SubscribeForWSMessages(suite.Cfg, connEvents, util.StartSendEvents, fmt.Sprintf(eventFilterTemplate, featureID))
+	defer util.UnsubscribeFromWSMessages(suite.Cfg, connEvents, util.StopSendEvents)
+	require.NoError(suite.T(), err, "error subscribing for WS ditto events")
+	for startID, path := range requestedFiles {
+		_, err := util.ExecuteOperation(suite.Cfg, suite.FeatureURL, operationStart, TestUpload.requestUpload(startID, path))
+		require.NoErrorf(suite.T(), err, msgErrorExecutingOperation, operationStart)
+	}
+
+	statuses := []interface{}{}
+	err = util.ProcessWSMessages(suite.Cfg, connEvents,
+		func(msg *protocol.Envelope) (bool, error) {
+			if (msg.Topic.String() == topicCreated || msg.Topic.String() == topicModified) && msg.Path == pathLastUpload {
+				statuses = append(statuses, msg.Value)
+				return IsTerminal(msg.Value, client.StateSuccess, client.StateFailed, client.StateCanceled), nil
+			}
+			return true, fmt.Errorf(msgUnexpectedValue, msg.Value)
+		})
+	require.NoError(suite.T(), err, "error processing upload status events")
+
+	lastUploadProgress := 0
+	for ind, status := range statuses {
+		uploadStatus := uploadStatus{}
+		err := util.Convert(status, &uploadStatus)
+		require.NoErrorf(suite.T(), err, "cannot convert %v to upload status", status)
+		suite.T().Logf("upload status event(%v)", uploadStatus)
+		require.GreaterOrEqual(suite.T(), uploadStatus.Progress, lastUploadProgress,
+			"upload status progress should be non-decreasing")
+		require.LessOrEqual(suite.T(), uploadStatus.Progress, 100,
+			"upload status progress should be less than or equal to 100%")
+		lastUploadProgress = uploadStatus.Progress
+		if ind < len(statuses)-1 {
+			require.Equal(suite.T(), client.StateUploading, uploadStatus.State, "wrong transitional upload state")
+		} else {
+			require.Equal(suite.T(), client.StateSuccess, uploadStatus.State, "wrong final upload state")
+		}
+	}
+}
+
+// IsTerminal checks if a status "state" property is contained in the specified states
+func IsTerminal(status interface{}, terminalStates ...string) bool {
+	if props, ok := status.(map[string]interface{}); ok {
+		state := props["state"]
+		for _, terminal := range terminalStates {
+			if state == terminal {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+// File/Directory test util functionalities start here
```

Comment:
> I am a little bit confused on all these util names around. There are:
> 
> * http_upload_test_util
> * upload_file_util
> * upload_test_util
> 
> Is it possible somehow to align/minimize this util crew?

They are moved here, because of that.